### PR TITLE
Make linting less badder

### PR DESCRIPTION
**Pull request by acolytec3**
This reimagines our lint process and hopefully makes it at least 2-3x faster

- Replace `prettier` with `biome` (almost exactly conforms to prettier standards but rust so therefore faster...maybe) for formatting and remove this from the `eslint` config
- Disables a few eslint rules that duplicate existing typescript functionality
  - `no-undef` (Typescript will catch any use of real undefined stuff.  This was just flagging for us referencing `NodeJS` in `devp2p`
  - `import/named`/`import/namespace`/`import/default` are all turned off - `typescript-eslint` [explicitly suggests disabling these](https://typescript-eslint.io/troubleshooting/typed-linting/performance#eslint-plugin-import)
 - Moves all package specific config overrides into the main eslint configuration file
 - Only run `eslint` from the root of the monorepo.  This is the biggest win as our previous way of running the linter per package was causing dramatic slowdowns since it was basically starting up the linter and parsing the file tree for each package
 - Removes all `prettier` and `tsconfig.lint.json` package specific config files as these are no longer needed
- Update the `precommit` and `prepush` git hooks to use the new linting setup
This is an alternative to #3792 where we replace prettier with `biome` just for formatting and then run the linter from the directory root rather than from each package (which means we only start the linter up once instead 13 times).  Together, these have reduced the time to lint the monorepo by 2.5x on my slow-mo Intel i9 processor from 2022.

### CODE REVIEW

**Review by holgerd77 — APPROVED** [review]
LGTM

**Review by jochem-brouwer — COMMENTED** [review]
2 questions. It also seems the style we have in the configs is now changed, since a lot of files got committed which seem to have lint changes?

File: config/eslint.cjs [hunk 1]
```diff
@@ -103,26 +108,53 @@ module.exports = {
...
+        'no-console': 'off',
+        '@typescript-eslint/no-unused-vars': 'off',
+      },
+    },
+    {
+      files: ['packages/statemanager/src/**', 'packages/vm/src/**', ],
+      rules: {
+        '@typescript-eslint/no-use-before-define': 'off',
+        'no-invalid-this': 'off',
+        'no-restricted-syntax': 'off',
```
[the remark this answers is not in the document]
Comment (acolytec3) [reply config/eslint.cjs:150]: I carried these over from the previous package-specific lint configs.  We can go through them and see if they can be removed.

File: config/cli/lint-diff.sh [hunk 2]
```diff
@@ -2,9 +2,9 @@
 REMOTE=$(git rev-parse --symbolic-full-name --abbrev-ref @{u})
 
 if [ -z "$REMOTE" ]; then
-    FILESCHANGED=". --ext .js,.jsx,.ts,.tsx"
+    FILESCHANGED=". --ext .js,.ts"
```
[the remark this answers is not in the document]
Comment (acolytec3) [reply config/cli/lint-diff.sh:5]: Yes, we don't use `tsx` and `jsx` in our code base.  Unless we start writing front-end web apps, we never will, so no point in checking for them.

**Comment by acolytec3** [comment]
> 2 questions. It also seems the style we have in the configs is now changed, since a lot of files got committed which seem to have lint changes?

I think there was maybe one subtle difference between prettier and biome and that accounts for most of the file changes.  It was just a little bit of formatting difference (e.g. adding a new line between closing brackets and opening of a new line sort of thing).

**Pull request merged**